src/types.ts

```diff
@@ -27,37 +27,69 @@ export interface Instance {
     account: string,
     bid: Bid
   ) => Promise<boolean>;
-  isZAuctionApprovedToTransferNft: (account: string) => Promise<boolean>;
-  getZAuctionSpendAllowance: (account: string) => Promise<ethers.BigNumber>;
+  isZAuctionApprovedToTransferNft: (
+    account: string,
+    contractAddress: string
+  ) => Promise<boolean>;
+  isZAuctionApprovedToTransferNftLegacy: (
+    account: string,
+    contractAddress: string
+  ) => Promise<boolean>;
   getZAuctionSpendAllowanceByBid: (
     account: string,
     bid: Bid
   ) => Promise<ethers.BigNumber>;
-  getTradeTokenAddress: () => Promise<string>;
+  getZAuctionSpendAllowance: (
+    paymentTokenAddress: string,
+    account: string
+  ) => Promise<ethers.BigNumber>;
+  getZAuctionSpendAllowanceLegacy: (
+    account: string
+  ) => Promise<ethers.BigNumber>;
+  getDefaultPaymentToken: () => Promise<string>;
+  setDefaultPaymentToken: (
+    defaultTokenAddress: string,
+    signer: ethers.Signer
+  ) => Promise<ethers.ContractTransaction>;
+  getDomainNetworkPaymentToken: (networkId: string) => Promise<string>;
+  setDomainNetworkPaymentToken: (
+    networkId: string,
+    domainNetworkToken: string,
+    signer: ethers.Signer
+  ) => Promise<ethers.ContractTransaction>;
+  getPaymentTokenForDomain: (domainTokenId: string) => Promise<string>;
   approveZAuctionSpendTradeTokensByBid: (
     bid: Bid,
     signer: ethers.Signer
   ) => Promise<ethers.ContractTransaction>;
   approveZAuctionSpendTradeTokens: (
+    paymentTokenAddress: string,
     signer: ethers.Signer
   ) => Promise<ethers.ContractTransaction>;
-  approveZAuctionTransferNft: (
+  approveZAuctionSpendTradeTokensLegacy: (
     signer: ethers.Signer
   ) => Promise<ethers.ContractTransaction>;
   approveZAuctionTransferNftByBid: (
     bid: Bid,
     signer: ethers.Signer
   ) => Promise<ethers.ContractTransaction>;
+  approveZAuctionTransferNft: (
```

Comment:
Why would be exposing this to our end user? They shouldn't have to worry about providing a `domainContractAddress`.

If you want to create a helper function to approve zAuction to transfer an NFT, it should take the NFT (token id) and then internally figure out what contract it belongs to.